objects-revisited/main.js

```diff
@@ -0,0 +1,84 @@
+const business = {
+  opens: '9am',
+  closes: '5pm',
+  totalEmployees: 4,
+  daysOpen: ['M', 'T', 'W', 'Th', 'F'],
+  employees: {
+    john: {
+      position: 'Cashier',
+      daysOfWeekWorking: ['M', 'W', 'Th', 'F']
+    },
+    ted: {
+      position: 'Stocking',
+      daysOfWeekWorking: ['M', 'T', 'F']
+    },
+    jack: {
+      position: 'Manager',
+      daysOfWeekWorking: ['M', 'T', 'W', 'Th', 'F']
+    },
+    jasmine: {
+      position: 'Barista',
+      daysOfWeekWorking: ['M', 'W', 'Th']
+    },
+  }
+}
+
+function addWeekends() {
+  business.daysOpen.push('S');
+  business.daysOpen.push('Su');
+  for (const property in business.employees) {
+    business.employees[property].daysOfWeekWorking.push('S');
+    business.employees[property].daysOfWeekWorking.push('Su');
+  }
+}
+
+function addEmployees() {
+  const xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://jsonplaceholder.typicode.com/users');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', () => {
+    for (let index = 0; index < 4; index++) {
+      business.employees[xhr.response[index].name] = xhr.response[index];
+      business.employees[xhr.response[index].name]['daysOfWeekWorking'] = [];
+      let day;
+      for (let i = 0; i < 5; i++) {
+        day = business.daysOpen[Math.floor(Math.random() * 5)];
+        if (business.employees[xhr.response[index].name]['daysOfWeekWorking'].length < 5 || business.employees[xhr.response[index].name]['daysOfWeekWorking'].length < 1) {
+          if(!business.employees[xhr.response[index].name]['daysOfWeekWorking'].includes(day)) {
+            business.employees[xhr.response[index].name]['daysOfWeekWorking'].push(day);
+          }
+        }
+      }
+      business.employees[xhr.response[index].name]['daysOfWeekWorking'].push('S');
+      business.employees[xhr.response[index].name]['daysOfWeekWorking'].push('Su');
+    }
+    business.employees['Clementine Bauch']['title'] = 'Customer Service';
+    business.employees['Ervin Howell']['title'] = 'Salesman';
+    business.employees['Leanne Graham']['title'] = 'Accountant';
+    business.employees['Patricia Lebsack']['title'] = 'Technician';
```

Comment:
I wouldn't listen for specific names. What if the API changes the way they return the data and those users are no longer the first 4?